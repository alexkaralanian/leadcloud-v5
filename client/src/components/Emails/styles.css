/*.table {
  padding: 0;
}
*/

.emailAddress {
  word-wrap: break-word;
}

@media (max-width: 425px) {
  .emailTable {
    max-width: 130px;
  }
}
