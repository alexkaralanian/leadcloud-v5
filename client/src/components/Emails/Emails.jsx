import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import { Grid, Row, Col, Table } from "react-bootstrap";
import moment from "moment";
import Loading from "../Loading/Loading";
import { tableCell, emailAddress, emailTable } from "./styles.css";
// import { error, background, content } from "../../sharedStyles/styles.css";

const Emails = ({ emails, createContact, isFetching, emailError }) =>
  isFetching ? (
    <Loading />
  ) : (
    <Grid>
      <Row>
        <Col xs={12}>
          <Table striped>
            <thead>
              <tr>
                <th>Sender</th>
                <th>Subject</th>
                <th>Date</th>
              </tr>
            </thead>
            <tbody>
              {emails &&
                emails.map(email => {
                  return (
                    <tr key={email.id}>
                      <td className={emailTable} id="sender">
                        <a
                          onClick={() =>
                            createContact(email.emailAddress, email.name)
                          }
                        >
                          <span className={emailAddress}>{email.name}</span>
                        </a>
                      </td>

                      <td id="email-subject">
                        <Link to={`/email/${email.id}`}>{email.subject}</Link>
                      </td>

                      <td id="email-date">
                        {moment(email.date).format("ddd, M/D/YY h:mma")}
                      </td>
                    </tr>
                  );
                })}
            </tbody>
          </Table>
        </Col>
      </Row>

      {/*error ? (
        <Row>
          <Col xs={12}>
            <p className={error}>{emailError}</p>
          </Col>
        </Row>
      ) : null*/}
    </Grid>
  );

export default Emails;

Emails.propTypes = {
  emails: PropTypes.array.isRequired,
  createContact: PropTypes.func.isRequired,
  isFetching: PropTypes.bool.isRequired,
  emailError: PropTypes.string.isRequired
};

Emails.defaultProps = {
  emails: [],
  createContact: () => {},
  isFetching: false,
  emailError: ""
};
