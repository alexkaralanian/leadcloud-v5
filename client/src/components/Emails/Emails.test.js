import React from "react";
import { shallow, configure } from "enzyme";
import Emails from "./Emails";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

test("Emails Component renders correctly", () => {
  const component = shallow(<Emails />);
  expect(component).toMatchSnapshot();
});

