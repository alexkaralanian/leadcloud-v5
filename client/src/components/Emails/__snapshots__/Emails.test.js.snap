// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emails Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Emails
    createContact={[Function]}
    emailError=""
    emails={Array []}
    isFetching={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "container",
      "children": <Row
        bsClass="row"
        componentClass="div"
>
        <Col
                bsClass="col"
                componentClass="div"
                xs={12}
        >
                <Table
                        bordered={false}
                        bsClass="table"
                        condensed={false}
                        hover={false}
                        responsive={false}
                        striped={true}
                >
                        <thead>
                                <tr>
                                        <th>
                                                Sender
                                        </th>
                                        <th>
                                                Subject
                                        </th>
                                        <th>
                                                Date
                                        </th>
                                </tr>
                        </thead>
                        <tbody>
                                
                        </tbody>
                </Table>
        </Col>
</Row>,
      "componentClass": "div",
      "fluid": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "row",
        "children": <Col
          bsClass="col"
          componentClass="div"
          xs={12}
>
          <Table
                    bordered={false}
                    bsClass="table"
                    condensed={false}
                    hover={false}
                    responsive={false}
                    striped={true}
          >
                    <thead>
                              <tr>
                                        <th>
                                                  Sender
                                        </th>
                                        <th>
                                                  Subject
                                        </th>
                                        <th>
                                                  Date
                                        </th>
                              </tr>
                    </thead>
                    <tbody>
                              
                    </tbody>
          </Table>
</Col>,
        "componentClass": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "col",
          "children": <Table
            bordered={false}
            bsClass="table"
            condensed={false}
            hover={false}
            responsive={false}
            striped={true}
>
            <thead>
                        <tr>
                                    <th>
                                                Sender
                                    </th>
                                    <th>
                                                Subject
                                    </th>
                                    <th>
                                                Date
                                    </th>
                        </tr>
            </thead>
            <tbody>
                        
            </tbody>
</Table>,
          "componentClass": "div",
          "xs": 12,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bordered": false,
            "bsClass": "table",
            "children": Array [
              <thead>
                <tr>
                                <th>
                                                Sender
                                </th>
                                <th>
                                                Subject
                                </th>
                                <th>
                                                Date
                                </th>
                </tr>
</thead>,
              <tbody>
                
</tbody>,
            ],
            "condensed": false,
            "hover": false,
            "responsive": false,
            "striped": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th>
                                    Sender
                  </th>
                  <th>
                                    Subject
                  </th>
                  <th>
                                    Date
                  </th>
</tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                      Sender
</th>,
                    <th>
                      Subject
</th>,
                    <th>
                      Date
</th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Sender",
                    },
                    "ref": null,
                    "rendered": "Sender",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Subject",
                    },
                    "ref": null,
                    "rendered": "Subject",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Date",
                    },
                    "ref": null,
                    "rendered": "Date",
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [],
              "type": "tbody",
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": <Row
          bsClass="row"
          componentClass="div"
>
          <Col
                    bsClass="col"
                    componentClass="div"
                    xs={12}
          >
                    <Table
                              bordered={false}
                              bsClass="table"
                              condensed={false}
                              hover={false}
                              responsive={false}
                              striped={true}
                    >
                              <thead>
                                        <tr>
                                                  <th>
                                                            Sender
                                                  </th>
                                                  <th>
                                                            Subject
                                                  </th>
                                                  <th>
                                                            Date
                                                  </th>
                                        </tr>
                              </thead>
                              <tbody>
                                        
                              </tbody>
                    </Table>
          </Col>
</Row>,
        "componentClass": "div",
        "fluid": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": <Col
            bsClass="col"
            componentClass="div"
            xs={12}
>
            <Table
                        bordered={false}
                        bsClass="table"
                        condensed={false}
                        hover={false}
                        responsive={false}
                        striped={true}
            >
                        <thead>
                                    <tr>
                                                <th>
                                                            Sender
                                                </th>
                                                <th>
                                                            Subject
                                                </th>
                                                <th>
                                                            Date
                                                </th>
                                    </tr>
                        </thead>
                        <tbody>
                                    
                        </tbody>
            </Table>
</Col>,
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "children": <Table
              bordered={false}
              bsClass="table"
              condensed={false}
              hover={false}
              responsive={false}
              striped={true}
>
              <thead>
                            <tr>
                                          <th>
                                                        Sender
                                          </th>
                                          <th>
                                                        Subject
                                          </th>
                                          <th>
                                                        Date
                                          </th>
                            </tr>
              </thead>
              <tbody>
                            
              </tbody>
</Table>,
            "componentClass": "div",
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bordered": false,
              "bsClass": "table",
              "children": Array [
                <thead>
                  <tr>
                                    <th>
                                                      Sender
                                    </th>
                                    <th>
                                                      Subject
                                    </th>
                                    <th>
                                                      Date
                                    </th>
                  </tr>
</thead>,
                <tbody>
                  
</tbody>,
              ],
              "condensed": false,
              "hover": false,
              "responsive": false,
              "striped": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <tr>
                    <th>
                                        Sender
                    </th>
                    <th>
                                        Subject
                    </th>
                    <th>
                                        Date
                    </th>
</tr>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <th>
                        Sender
</th>,
                      <th>
                        Subject
</th>,
                      <th>
                        Date
</th>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Sender",
                      },
                      "ref": null,
                      "rendered": "Sender",
                      "type": "th",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Subject",
                      },
                      "ref": null,
                      "rendered": "Subject",
                      "type": "th",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Date",
                      },
                      "ref": null,
                      "rendered": "Date",
                      "type": "th",
                    },
                  ],
                  "type": "tr",
                },
                "type": "thead",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": "tbody",
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
